nuxeo-features/nuxeo-automation/nuxeo-automation-server/src/main/java/org/nuxeo/ecm/automation/server/jaxrs/adapters/AsyncOperationAdapter.java

```diff
@@ -0,0 +1,352 @@
+                int customHttpStatus = ((RestOperationException) unWrapException).getStatus();
+                throw new NuxeoException(exceptionMessage, cause, customHttpStatus);
+            }
+            throw new NuxeoException(exceptionMessage, cause);
+        }
+    }
+
+    @GET
+    @Path("{executionId}/status")
+    public Object status(@PathParam("executionId") String executionId)
+            throws IOException, URISyntaxException, MessagingException {
+        if (isCompleted(executionId)) {
+            String error = getError(executionId);
+            if (error != null) {
+                throw new NuxeoException(error, Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+            }
+            String resURL = StringUtils.stripEnd(ctx.getURL(), "/status");
+            return redirect(resURL);
+        } else {
+            Object result = "RUNNING";
+            if (isAsync(executionId)) {
+                Serializable commandId = getCommandId(executionId);
+                result = getAsyncService(executionId).getStatus(commandId);
+            }
+            return ResponseHelper.getResponse(result, request, Response.Status.OK.getStatusCode());
+        }
+    }
+
+    @GET
+    @Path("{executionId}")
+    public Object result(@PathParam("executionId") String executionId)
+            throws IOException, URISyntaxException, MessagingException {
+
+        if (isCompleted(executionId)) {
+            Object output = getResult(executionId);
+
+            String error = getError(executionId);
+
+            // cleanup after result is accessed
+            cleanup(executionId);
+
+            if (error != null) {
+                throw new NuxeoException(error, Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+            }
+
+            // if output has a "url" key assume it's a redirect url
+            if (output instanceof Map) {
+                Object url = ((Map) output).get("url");
+                if (url instanceof String) {
+                    String baseUrl = VirtualHostHelper.getBaseURL(ctx.getRequest());
+                    return redirect(baseUrl + url);
+                }
+            }
+            return ResponseHelper.getResponse(output, request, Response.Status.OK.getStatusCode());
+        }
+
+        throw new WebResourceNotFoundException("Execution with id=" + executionId + " not found");
+    }
+
+    @DELETE
+    @Path("{executionId}")
+    public Object abort(@PathParam("executionId") String executionId) throws IOException, MessagingException {
+        if (exists(executionId) && !isCompleted(executionId)) {
+            // XXX: only supports abort of async commands
+            if (isAsync(executionId)) {
+                Serializable commandId = getCommandId(executionId);
+                return getAsyncService(executionId).abort(commandId);
+            }
+            return ResponseHelper.getResponse("RUNNING", request, Response.Status.OK.getStatusCode());
```

Comment:
I would go for an error case. We shouldn't be allowed to abort something not async when using async adapter.

Reply:
the operation is executed asynchronously with the @async adapter but for now we don't stop the thread that is running it and it returns a RUNNING status in this case, 
only the operation of type async (a bulk action operation) can be aborted for now